pcm-core.cpp

```diff
@@ -79,6 +79,7 @@ extern "C" {
 	int pcm_c_init()
 	{
 		PCM * m = PCM::getInstance();
+		regs = (EventSelectRegister*)malloc(sizeof(EventSelectRegister) * m->getMaxCustomCoreEvents());
```

Comment:
unfotunately regs is already accessed before pcm_c_init call in pcm_c_build_core_event (see pcm.so/c_example.c). As a simple fix I would rather suggest to 
introduce
#define PCM_MAX_FIXED_COUNTERS               (3) 
#define PCM_MAX_CUSTOM_COUNTERS               (8)
and redefine
#define PERF_MAX_COUNTERS               (PCM_MAX_FIXED_COUNTERS + PCM_MAX_CUSTOM_COUNTERS)
in types.h
just replace EventSelectRegister regs[4]; to EventSelectRegister regs[PCM_MAX_CUSTOM_COUNTERS]; in pcm-core.cpp

Reply:
Sorry I didn't notice that `regs` is used before `pcm_c_init` is called. Fixed as suggested.